[Subset of upstream patch that applies to TigerVNC 1.1 and is needed to
 silence Solaris Studio C++ compiler errors of: "TXWindow.cxx", line 93:
 Error: Narrowing conversion of 'int' value to 'char' is not allowed here.]

From a9af1f12466e201fed5a72bbba2607fb55f7e042 Mon Sep 17 00:00:00 2001
From: Pierre Ossman <ossman@cendio.se>
Date: Wed, 17 Jun 2015 10:47:28 +0200
Subject: [PATCH] Fix some bad signed/unsigned mismatches

---
 common/rfb/VNCSConnectionST.h | 2 +-
 unix/tx/TXWindow.cxx          | 8 ++++----
 vncviewer/parameters.cxx      | 4 ++--
 win/rfb_win32/SInput.cxx      | 2 +-
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git unix/tx/TXWindow.cxx unix/tx/TXWindow.cxx
index 1f69702..a681917 100644
--- unix/tx/TXWindow.cxx
+++ unix/tx/TXWindow.cxx
@@ -89,11 +89,11 @@ void TXWindow::init(Display* dpy, const char* defaultWindowClass_)
   XSetFont(dpy, defaultGC, defaultFont);
   XSelectInput(dpy, DefaultRootWindow(dpy), PropertyChangeMask);
 
-  static char dotBits[] = { 0x06, 0x0f, 0x0f, 0x06};
-  dot = XCreateBitmapFromData(dpy, DefaultRootWindow(dpy), dotBits,
+  static unsigned char dotBits[] = { 0x06, 0x0f, 0x0f, 0x06};
+  dot = XCreateBitmapFromData(dpy, DefaultRootWindow(dpy), (char*)dotBits,
                               dotSize, dotSize);
-  static char tickBits[] = { 0x80, 0xc0, 0xe2, 0x76, 0x3e, 0x1c, 0x08, 0x00};
-  tick = XCreateBitmapFromData(dpy, DefaultRootWindow(dpy), tickBits,
+  static unsigned char tickBits[] = { 0x80, 0xc0, 0xe2, 0x76, 0x3e, 0x1c, 0x08, 0x00};
+  tick = XCreateBitmapFromData(dpy, DefaultRootWindow(dpy), (char*)tickBits,
                                tickSize, tickSize);
   defaultWindowClass = rfb::strDup(defaultWindowClass_);
 }
